docs/file-upload-process.md

```diff
@@ -0,0 +1,107 @@
+
+### LIFE-CYCLE OF AN ENVELOPE - [see state diagram](../resources/images/envelope-life-cycle.png)
+
+#### 1. Create an envelope
+The first step is to create an envelope. The following endpoint is called and the envelope is created (an envelopeId is generated and returned in the responseHeader) : -
+
+- POST       /file-upload/envelopes
+
+[click here for more details](https://github.com/hmrc/file-upload#create-an-envelope)
+
+
+An envelope can contain one or more files. Once created, an envelope will be in an OPEN state, and will remain in this state until a “routing request” is sent.
+
+#### 2. Upload file(s)
+After the envelope has been created, the client can now send files using the below endpoint: -
+
+- POST        /file-upload/upload/envelopes/{envelope-id}/files/{file-id}
+
+envelope-id - the ID that was returned in step 1
+file-id - a user generated value. This can be any value the use wishes (file-id’s must be unique within an envelope). One request per file.
+
+Files are uploaded to the QUARANTINE bucket and then virus-scanned: -
+- if no issues are found, the file is moved to the TRANSIENT bucket (and file status is set to AVAILABLE)
+- if the file is deemed to be INFECTED, the file remains in the QUARANTINE bucket (client will be notified if a callbackUrl was provided)
+
+[click here for more details](https://github.com/hmrc/file-upload-frontend#upload-file)
+
+
+#### 3. Send routing request
+ Once all files have been uploaded, the client can send a “routing request”: - 
+
+- POST       /file-routing/requests
+
+Once this has been sent, the following will happen: -
+ - the envelope moves into a SEALED state 
+ - and the client can no longer upload files to the given envelope
+ - the envelope moves to a CLOSED state
+
+[click here for more details](https://github.com/hmrc/file-upload#create-file-routing-request)
+
+
+#### 4. Delete envelope
+The envelope will remain in a CLOSED state until the envelope is DELETED. The envelope is deleted with the following endpoint
+
+DELETE     /file-transfer/envelopes/{envelope-id}
+envelope-id - the identifier for the envelope to be deleted (this is a soft delete)
+
+[click here for more details](https://github.com/hmrc/file-upload#soft-delete-an-envelope)
+
+
+#### Envelope Statuses
+See [here](https://github.com/hmrc/file-upload#envelope-statuses) for envelope statuses
+
+### SINGLE FILE PROCESS
+Clients that aim to upload a single file and have no need for "routing", can follow the following steps: -
+1. [create an envelope](https://github.com/hmrc/file-upload#create-an-envelope)
+2. [upload file](https://github.com/hmrc/file-upload-frontend#upload-file)
+3. [download file](https://github.com/hmrc/file-upload#download-file)
+
```

Comment:
Files can be deleted as well: {{be-url}}/envelopes/{{envelopeId}}/files/{{fileId}}

Reply:
good point, will add that